app.rb

```diff
@@ -0,0 +1,161 @@
+require_relative 'person'
+require_relative 'student'
+require_relative 'classroom'
+require_relative 'rentals'
+require_relative 'teacher'
+
+class App
+  def initialize
+    @rentals = []
+    @people = []
+    @books = []
+  end
+
+  def list_all_people
+    puts 'These are all the people in the library:'
+    if @people.empty?
+      puts 'Nobody visited the library'
+    else
+      @people.each do |person|
+        puts "Id: #{person.id}, Name: #{person.name}, Age: #{person.age}, Class: #{person.class}"
+      end
+    end
+  end
+
+  def list_all_books
+    puts 'These are all the books in the library:'
+    if @books.empty?
+      puts 'No book is in the library'
+    else
+      @books.each do |book|
+        puts " Author: #{book.author}, Title: #{book.title}"
+      end
+    end
+  end
+
+  def create_person
+    print 'Press [1] to create a teacher or press [2] to creat a student'
+    input = gets.chomp.to_i
+    if input == 1
+      create_teacher
+    elsif input == 2
+      create_student
+    else
+      'Invalid. Choose [1] to create a teacher or [2] to create a student'
+    end
+  end
+
+  def create_teacher
+    print 'Enter name of the teacher:'
+    name = gets.chomp
+    print 'Enter the age of the teacher:'
+    age = gets.chomp.to_i
+    print 'Enter his/her specialization:'
+    specialization = gets.chomp
+    teacher = Teacher.new(age, specialization, name)
+    @people << teacher
+    puts 'Teacher was created successfully'
+  end
+
+  def create_student
+    print 'Enter the name of the student:'
+    name = gets.chomp
+    print 'Enter the age of the student:'
+    age = gets.chomp
+    print 'Does the student have parent permission? [Y/N]:'
+    parent_permission = gets.chomp
+    if parent_permission.downcase == 'y'
+      student = Student.new('classroom', name, false, age)
+    elsif parent_permission.downcase == 'n'
+      student = Student.new('classroom', name, true, age)
+    else
+      puts 'Invalid input'
+      return
+    end
+    @people << student
+    puts 'Student created successfully'
+  end
+
+  def create_book
+    print 'Enter the title of the book:'
+    title = gets.chomp
+    print 'Enter name of author'
+    author = gets.chomp
+    book = Book.new(title, author)
+    @books << book
+    puts 'Book created successfully'
+  end
```

Comment:
- [ ] While creating a book, I'm getting the below error, Please make sure to fix this and users should be able to create and display all entities without any error. (Also getting errors while creating student)

![image](https://github.com/mosidrum/OOP-school-library/assets/83803414/5bd152ce-bb19-485f-89d3-ed1e9cf3d60c)